pyteal/ast/router.py

```diff
@@ -259,13 +266,203 @@ class CondNode:
 CondNode.__module__ = "pyteal"
 
 
-@dataclass
+@dataclass(frozen=True)
+class CondWithMethod:
+    """CondWithMethod unifies the generation to CondNode from compilation w/wo frame pointer.
+
+    It takes following arguments:
+    - method_sig: str
+    - condition: Expr | int, which matches the on_completion conditions
+    - method: ABIReturnSubroutine, assuming that the method is ABI routable
+    """
+
+    method_sig: str
+    condition: Expr | int
+    method: ABIReturnSubroutine
+
+    def to_cond_node(self, use_frame_pt: bool = False) -> CondNode:
+        walk_in_cond = Txn.application_args[0] == MethodSignature(self.method_sig)
+
+        if not (isinstance(self.condition, Expr) or self.condition == 1):
```

Comment:
1 is an enum?

Reply:
no, `self.condition` can be `Expr | int`:
- `Expr` stands for some on-call conditions, e.g.,:
  `Txn.on_completion() == OnComplete.NoOp and Txn.on_completion() == OnComplete.OptIn`.
- 1 or 0 stands for 2 extreme cases of on-call conditions:
  - 1 stands for allowed for any on call option
  - 0 stands for no on call option is allowed, which is a dead branch.

so here, it means if `self.condition` is not valid, do something. Validity decided by `Expr` or == 1.

Reply:
also explained in more documentation [52a9cd6](https://github.com/algorand/pyteal/pull/600/commits/52a9cd6335ded1cf340f492c7b2cbf23bdadfe98)